packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -35,8 +35,24 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
 
   if (process.client) {
     const callback = () => { writeClientCookie(name, cookie.value, opts as CookieSerializeOptions) }
+
     if (opts.watch) {
-      watch(cookie, callback, { deep: opts.watch !== 'shallow' })
+      const channel = new BroadcastChannel(`nuxt:cookies:${name}`)
+      onUnmounted(() => { channel.close() })
+      let watchPaused = false
+
+      watch(cookie, (newValue) => {
+        if (watchPaused) { return }
+        callback()
+        channel.postMessage(newValue)
```

Comment:
I think we probably want to post the message within callback rather than in the watcher. I think we still want to broadcast cookie changes even if we don't have `opts.watch` enabled.